/*
 * This file was automatically generated by EvoSuite
 * Sat May 24 20:01:54 GMT 2025
 */

package original;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import original.Solution3209;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class Solution3209_ESTest extends Solution3209_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      Solution3209 solution3209_0 = new Solution3209();
      int[] intArray0 = new int[5];
      intArray0[0] = (-1);
      long long0 = solution3209_0.countSubarrays(intArray0, (-1));
      assertEquals(1L, long0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      Solution3209 solution3209_0 = new Solution3209();
      int[] intArray0 = new int[5];
      long long0 = solution3209_0.countSubarrays(intArray0, (-958));
      assertEquals(0L, long0);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      Solution3209 solution3209_0 = new Solution3209();
      // Undeclared exception!
      try { 
        solution3209_0.countSubarrays((int[]) null, 563);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("original.Solution3209", e);
      }
  }
}
